src/sqlparser/src/ast/mod.rs

```diff
@@ -106,19 +110,57 @@ impl Ident {
         }
     }
 
-    /// Create a new quoted identifier with the given quote and value. This function
-    /// panics if the given quote is not a valid quote character.
-    pub fn with_quote<S>(quote: char, value: S) -> Self
+    /// Create a new identifier from word
+    pub fn new_from_word(w: &Word) -> Result<Ident, ParserError> {
+        if w.value.is_empty() {
+            Err(ParserError::ParserError(format!(
+                "zero-length delimited identifier at or near \"{w}\""
+            )))
+        } else {
+            Ok(Ident {
+                value: w.value.clone(),
+                quote_style: w.quote_style,
+            })
+        }
+    }
```

Comment:
Since `new_from_word` is only used in `Word::to_ident`, I think we can simply put the implementation there.

Reply:
removed implementation to Word::to_ident